hvplot/converter.py

```diff
@@ -1819,11 +1828,94 @@ def method_wrapper(ds, x, y):
                 threshold=self.kwds.get('threshold', 0.5),
             )
 
+        # a workaround to show hover info for datashaded points
+        if self.hover and self.datashade and self.kind == 'points':
+            if self.hover_mode != 'mouse':
+                param.main.param.warning(
+                    f'Got unsupported hover_mode={self.hover_mode!r} for '
+                    f"datashaded points; reverting to 'mouse'."
+                )
+
+            stream = PointerXY
+            if len(self.data) > 10000:
+                stream = Tap
+                param.main.param.warning(
+                    'Hovering over datashaded points is slow for large datasets; '
+                    'tap on the plot to see a hover tooltip over desired point.'
+                )
+
+            inspector = inspect_points.instance(
+                streams=[stream], transform=self._datashade_hover_transform
+            )
+            processed *= inspector(processed).opts(
+                size=10,
+                alpha=0,
+                tools=['hover'],
+                hover_mode=self.hover_mode,
+                hover_tooltips=self.hover_tooltips,
+                hover_formatters=self.hover_formatters,
+            )
+
         opts = filter_opts(eltype, dict(self._plot_opts, **style), backend='bokeh')
         layers = self._apply_layers(processed).opts(eltype, **opts, backend='bokeh')
         layers = _transfer_opts_cur_backend(layers)
         return layers
 
+    def _datashade_hover_transform(self, df):
+        if not len(df):
+            return df
+
+        # show at least the x and y columns
+        cols = self.hover_cols.copy()
+        if self.x not in cols:
+            cols.append(self.x)
+        if self.y not in cols:
+            cols.append(self.y)
+
+        # handle aggregator, e.g. ds.sum('column') or ds.count_cat('column')
+        agg_col = None
+        agg_series_map = {}
+        if self.aggregator and not isinstance(self.aggregator, str) and self.aggregator.column:
+            agg_col = self.aggregator.column
+            agg_op = type(self.aggregator).__name__
+            if hasattr(df, agg_op):  # df.sum/df.count
+                agg_value = df.agg({agg_col: agg_op})
+            elif agg_op == 'count_cat':
+                agg_value = df[agg_col].value_counts()
+
+            if agg_col in cols:
+                cols.remove(agg_col)
+        else:
+            key = 'Count'
+            for i in range(1, 10):
+                if key in df.columns:
+                    key = f'Count_{i}'
+                else:
+                    break
+            agg_value = pd.Series([len(df)], index=[key])
+
+        # take the mean of numeric columns
```

Comment:
Is the rest of the method generated with AI? If so, please clearly state so.  

Otherwise, please explain _why_ you take the mean. In general, comments should not explain _what_ is done but _why_ it is done.

Reply:
No it's not AI. I decided to add comments since the code was getting chunky, and easier to navigate.

> Comments should not explain what is done but why it is done.

Sometimes it's easier to read English than code and guidelines don't necessarily have to be followed all the time

Reply:
Sorry about that. It just had the structure of an AI output. 

> ... guidelines don't necessarily have to be followed all the time.

I agree, but the use of `mean` need to be explained, either in the code or in the original comment. 

This can return an output that does not exist in the original DataFrame when more points are in the input DataFrame. For example, the two close points in your example will return 350 for the population when they are close to each other. Is this okay? Maybe, but you need to state that you have made this decision.

Reply:
@ahuang11 , @hoxbro thinks you are so intelligent that it must be artificial! :-)